### PR TITLE
DeckPicker: disable functionality during storage migration

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
A subset of functionality may be problematic while we are migrating storage.

## Fixes
- Related: https://github.com/ankidroid/Anki-Android/issues/13094
- Related: #5304

## Approach
* Show a snackbar when selected functionality is active
* Do not disable 'restore from backup' functionality
* ⚠️ DO not grey out the menu items
  * If they can be disabled with an explanation when clicked on, I'd support this - haven't looked into it

## How Has This Been Tested?
Manually: API 31 AOSP emulator

Screenshot 2023-02-27 at 00 46 55
Screenshot 2023-02-27 at 00 46 42

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.kt [hunk 1]
```diff
@@ -161,6 +162,11 @@ open class CollectionTask<Progress, Result>(val task: TaskDelegateBase<Progress,
                 Pair(true, result)
             }
         }
+
+        companion object {
+            fun canCheckDatabase(context: Context): Boolean =
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.kt:167]: *Currently*, I don't see a reason to create this method. Unless you already have something in mind to add into this method, I suggest inlining it.

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/MediaService.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        showSnackbar(R.string.functionality_disabled_during_storage_migration, Snackbar.LENGTH_SHORT)
+        return null
+    }
+
+    return withProgress(R.string.check_media_message) {
+        CollectionManager.withCol { media.performFullCheck() }
+    }
+}
+
+fun canCheckMedia(context: Context) = !ScopedStorageService.userMigrationIsInProgress(context)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/MediaService.kt:39]: same here

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 3]
```diff
@@ -1100,8 +1100,8 @@ open class DeckPicker :
         val autoSyncIsEnabled = preferences.getBoolean("automaticSyncMode", false)
         val syncIntervalPassed = TimeManager.time.intTimeMS() - lastSyncTime > AUTOMATIC_SYNC_MIN_INTERVAL
         val isNotBlockedByMeteredConnection = preferences.getBoolean(getString(R.string.metered_sync_key), false) || !isActiveNetworkMetered()
-
-        if (isLoggedIn() && autoSyncIsEnabled && NetworkUtils.isOnline && syncIntervalPassed && isNotBlockedByMeteredConnection) {
+        val canSync = canSync(this)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:1103]: same here. Method name is especially misleading here, so definetely worth inling this

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.kt:167]: It's used in 3 places - unsure on this one, but OK

**Review by BrayanDSO — APPROVED** [review]

**Pull request merged**